### PR TITLE
Tweak readme.md

### PR DESCRIPTION
**Pull request by papb**
This PR relates to issue #55. In particular, to the first bullet of [this comment I made there](https://github.com/SamVerschueren/tsd/issues/55#issuecomment-616241488).

I'm not even sure if what I've written in this PR is correct; but it is what I understood after a lot of digging. If it's wrong, please explain and then I will be happy to fix the PR.

I am pretending to be writing for my past self, which would have really wanted some explanation in this direction.

Thanks!!

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 > Check TypeScript type definitions
 
+## Overview
+
+This tool lets you write tests for your type definitions (i.e. your `.d.ts` files) by creating files with the `.test-d.ts` extension.
+
+These `.test-d.ts` files will not be executed, and not even compiled in the standard way. Instead, these files will be magically parsed for special constructs such as `expectError<Foo>(bar)` and then statically analyzed against your type definitions.
```
Comment (Ethan-Arrowood) [inline readme.md:9]:
```suggestion
These `.test-d.ts` files will not be executed, and not even compiled in the standard way. Instead, these files will be parsed for special constructs such as `expectError<Foo>(bar)` and then statically analyzed against your type definitions.
```

This is a good summary imo! Just one change to remove the word `magically` and it lgtm 😄

Comment (papb) [reply readme.md]: @Ethan-Arrowood Thanks! Suggestion accepted!

File: readme.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 > Check TypeScript type definitions
 
+## Overview
```
Comment (SamVerschueren) [inline readme.md:5]: Can you move this below `## Install`? And leave 2 blank lines between the sections?

Comment (papb) [reply readme.md]: @SamVerschueren done!

**Comment by SamVerschueren** [comment]
Thanks! :)

**Pull request merged**